Deeploy/CommonExtensions/DataTypes.py

```diff
@@ -76,10 +76,26 @@ class uint64_t(IntegerImmediate):
     signed = False
 
 
+# BFloat16 in PULP systems
+class float16alt(FloatImmediate):
```

Comment:
Sure, but it's not very clear to me whether the `typeName` should align with the actual data type used in C. In the review to Alberto's PR you suggested to align the class name and the `typeName`, but that forces us to use a nondescript `float16alt` as class name to align with the data type used by PULP compilers.
That is also the reason why I did not yet fix the typing test.

If `typeName` does not need to be aligned with what backend data type, I'd prefer to use `bfloat16_t`. Otherwise, I guess we can also redefine the type, but where should that be done inside Deeploy?

Reply:
Ah okay, I didn't realize. I would not suggest naming the class differently from the `typeName`, as I find it confusing. That said, I think whichever way we go (without modifying any compilers) there is a bit surface area for confusion. I leave it up to you to decide. I like having the same type names in Deeploy and C (makes it easier to grep for these things), so my advice would be to keep the current solution, if it aligns with C nomenclature.

Reply:
For me the best is to realign to a `bfloat16_t` & co typename (after all, we already use `typedef`'d types for integers, not the native `int`, `char`, etc.).
So I am introducing a dedicated `types.h` header included from the `DeeployTARGETMath.h` headers.

Reply:
Also, the intent of `bfloat16_t` is definitely clearer than that of `float16alt`...

Reply:
Agreed!